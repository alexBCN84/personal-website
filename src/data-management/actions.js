export const SET_ACTIVE_LANGUAGE = 'set language';