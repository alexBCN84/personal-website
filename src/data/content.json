
{
    "ES": {
        "kicker": "bienvenid@ a mi web!",
        "headline": "Soy traductor de formación, profesor de vocación y desde 2016 desarrollador web",
        "description": "He pasado gran parte de mi vida entre España, Francia, Reino Unido y Estados Unidos. Actualmente vivo en Berlín, donde trabajo como Ingeniero Front-End en Zalando SE",
        "aboutChevronText": "conócenos",
        "cvCTA": {
            "text": "Ver CV",
            "href": "https://alexgines.netlify.app/"
        }
    },
    "EN": {
        "kicker": "Welcome to my Site",
        "headline": "I am a former translator, a teacher at heart, and a Web Developer since 2016",
        "description": "I've spent most of my life between Spain, France, the United Kingdom and the United States. I'm currently based in Berlin, where I work as a Frontend Engineer at Zalando SE",
        "aboutChevronText": "about us",
        "cvCTA": {
            "text": "See my CV",
            "href": "https://alexgines.netlify.app/"
        }
    }
}