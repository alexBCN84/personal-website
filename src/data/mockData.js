export const rowMockData = {
    "file": {
      "childImageSharp": {
        "fixed": {
          "base64": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAUCAYAAACNiR0NAAAACXBIWXMAAC4jAAAuIwF4pT92AAACU0lEQVQ4y22Vz0tVURDH73u+vBFYIYhERItq3x8QEQgWZYsMikDoJ5VRq2zRxnDjRpAgqk3Zs0KJeBUtoq3VIgoKKgrLhH4aRLnQRT/o3T5D35Hxdg98mTnnzHzPnJk59yZZliU2TIJyooG+uV6vV5ET4JfwFgyDjuBXjhyRrGQ6JG3oL7N/4zcYY+0q8gq4J2Ib46y3B///yZAnZfwZ7MGhKckN1heBLvBBtn3zSHEqRzLmo4g0R3IAdIOdYKHWFoCqfPp0u7I7tOm0kUCSymit9h7psOW5wy5of2tcfIXxJydBNoTIR+RwRgVpdRvJkor1DlRsYYscdsugEnK6Rlc6jdgFbqG3eM7QK9K3i2OHLQ6rck3hRD/9rAxXgb3gJmgJ7eK3SMF3cMMmE5COFVS8VWRDmh/DruaEBdW/AyZN+Wm5Caf6NXpFeFQRHrEIQLMqvEJYKfvBTBFFQr/qYiXZx0ewX1c2wrta/2opVludcsLX4L4XRPKQHE6AjWAdOBwi7AT94JvsrLWuWbOb80XwBywJRXljzy+XI7tyzftQVbfRo/k0uJ0oAhsHtdEZ24hC+MvoRr8ufbVsHmjeoXmXn/5M+bBkPwYzoNEjDhE+gdTyOKXcL9PeOPgy92RR1uuEGg5DVoDQ5N7Em5CXdLUpf2rIc3NNnatsjzYux++iPUGbI0vKbxryel4+g952+W/hcRlY9fYxX1rQwKme2qSe5UDYK/wmbgDPRfwDPFTUVX1gZ7X3HmwLZKV5rCJtCFdtB5azpyqYRf0CjKoTKkW/gL+tuhI3CufaNQAAAABJRU5ErkJggg==",
          "width": 40,
          "height": 40,
          "src": "/static/b0ddc18a1e874e013ee42035351d5925/ff4be/logo.png",
          "srcSet": "/static/b0ddc18a1e874e013ee42035351d5925/ff4be/logo.png 1x,\n/static/b0ddc18a1e874e013ee42035351d5925/183c2/logo.png 1.5x,\n/static/b0ddc18a1e874e013ee42035351d5925/829e2/logo.png 2x"
        }
      }
    },
    "allDataJson": {
      "edges": [
        {
          "node": {
            "ES": {
              "kicker": "bienvenid@ a mi web!",
              "headline": "Soy traductor de formación, profesor de vocación y desde 2016 desarrollador web",
              "description": "He pasado gran parte de mi vida entre España, Francia, Reino Unido y Estados Unidos. Actualmente vivo en Berlín, donde trabajo como Ingeniero Front-End en Zalando SE",
              "aboutChevronText": "conócenos",
              "cvCTA": {
                "text": "Ver CV",
                "href": "https://alexgines.netlify.app/"
              }
            },
            "EN": {
              "kicker": "Welcome to my Site",
              "headline": "I am a former translator, a teacher at heart, and a Web Developer since 2016",
              "description": "I've spent most of my life between Spain, France, the United Kingdom and the United States. I'm currently based in Berlin, where I work as a Frontend Engineer at Zalando SE",
              "aboutChevronText": "about us",
              "cvCTA": {
                "text": "See my CV",
                "href": "https://alexgines.netlify.app/"
              }
            }
          }
        }
      ]
    }
  }

  export const initialStateMock = {
    languages: { active: 'EN', inactive: 'ES' }
};