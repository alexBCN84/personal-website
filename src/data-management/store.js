import React, { createContext, useReducer } from 'react';
import { graphql, useStaticQuery } from 'gatsby';
import { SET_ACTIVE_LANGUAGE } from './actions';

const initialState = {
    languages: { active: 'EN', inactive: 'ES' }
};

const store = createContext(initialState);
const { Provider } = store;

const DataProvider = ({ children }) => {
    const [state, dispatch] = useReducer((state, action) => {
        switch (action.type) {
            case SET_ACTIVE_LANGUAGE:
                return {...state, languages: action.languages}
            default:
                throw new Error();
        }
    }, initialState)
    const data = useStaticQuery(
        graphql`
          query {
            file(relativePath: { eq: "logo.png" }) {
              childImageSharp {
                fixed(width: 40, height: 40) {
                  ...GatsbyImageSharpFixed
                }
              }
            }
            allDataJson {
              edges {
                node {
                  ES {
                    kicker
                    headline
                    description
                    aboutChevronText
                    cvCTA {
                      text
                      href
                    }
                  }
                  EN {
                    kicker
                    headline
                    description
                    aboutChevronText
                    cvCTA {
                      text
                      href
                    }
                  }
                }
              }
            }
          }
        `
      );
      const logo = data.file.childImageSharp.fixed
      const { headline, kicker, description, cvCTA, aboutChevronText: about } = data?.allDataJson.edges[0].node[
        state?.languages.active
      ];
      const heroBannerData = {logo, headline, kicker, description, cvCTA, about}
    return <Provider value={{heroBannerData, dispatch, state}}>{children}</Provider>
}

export {store, DataProvider};