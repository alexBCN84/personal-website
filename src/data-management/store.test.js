import { DataProvider, reducer } from './store';
import { render } from '@testing-library/react';
import React from 'react';
import { rowMockData, initialStateMock } from '../data/mockData';
import { SET_ACTIVE_LANGUAGE } from './actions';

function renderDataProvider(){
    return render(
        <DataProvider data={rowMockData}>
                <div>
                    <h1>Test header 1</h1>
                    <h2>Test header 2</h2>
                </div>
        </DataProvider>
    )
}

const useReducer = jest.spyOn(React, 'useReducer');
useReducer.mockImplementation(() => ([state, dispatch]));



describe('store', () => {
    it('renders DataProvider without crashing', () => {
        const { container } = renderDataProvider();
        expect.assertions(1);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('sets active language', () => {
        const action = {
            type: SET_ACTIVE_LANGUAGE,
            languages: { active: 'ES', inactive: 'EN' }
        }

        expect.assertions(1);
        expect(reducer(initialStateMock, action)).toEqual({languages: action.languages});
    });

    it('throws error if action type does not match any case', () => {
        const consoleError = console.error;
        // reset console Error to avoid error message being logged when running test
        console.error = jest.fn();

        const NO_CASE = "no case";
        const action = {
            type: NO_CASE,
            languages: { active: 'ES', inactive: 'EN' }
        }
        expect.assertions(1);
        expect(() => reducer(initialStateMock, action)).toThrow('Could not find a case match');
        // set console.error back to its original state
        console.error = consoleError;
    });
});