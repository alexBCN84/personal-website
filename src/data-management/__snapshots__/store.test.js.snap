// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store renders DataProvider without crashing 1`] = `
<div>
  <h1>
    Test header 1
  </h1>
  <h2>
    Test header 2
  </h2>
</div>
`;
